### ISSUE TITLE
GitHub code navigation support 

### ISSUE THREAD

**Issue by johnpryan** [dart-lang/sdk#37418]
GitHub uses [github.com/github/semantic](https://github.com/github/semantic) for [code navigation](https://help.github.com/en/articles/navigating-code-on-github) (beta). It would be great to have Dart on the list of supported languages.

**Comment by bwilkerson** [comment]
Agreed. Unfortunately the docs for semantic are not very encouraging:

> ... adding support for a new language is a nontrivial amount of work.

There's work being done at Microsoft to add a Kythe-like language service to support similar sorts of tooling (https://code.visualstudio.com/blogs/2019/02/19/lsif). This would be much easier to implement because we wouldn't need to re-write everything. I don't know whether GitHub has any plans to support this protocol.

**Comment by davidmorgan** [comment]
`package:pedantic` recommends `omit_local_variable_types` in large part because google internal tools make it easy enough to find types by following cross references.

It doesn't look like adding code navigation support is an easy thing to do; it does seem like it could give a lot of value. Please count 'reduces difference between external and internal development' as part of that value :)

**Comment by gnprice** [comment]
Related newer issue: https://github.com/dart-lang/sdk/issues/50523

It sounds like what GitHub now asks for in order to support a given language is [a tree-sitter parser](https://tree-sitter.github.io/tree-sitter/creating-parsers) for it, which is somewhat more encouraging.
